test/e2e/packagemanifest_e2e_test.go

```diff
@@ -3,170 +3,196 @@ package e2e
+	})
 
-		for _, v := range relatedImages {
-			_, ok := expectedRelatedImages[v]
-			require.True(GinkgoT(), ok, "Expect this image %s to exist in the related images list\n", v)
-		}
+	Context("Given an operator-registry image", func() {
+		Context("When a CatalogSource is created using gRPC catalog source type", func() {
+			When("Package manifest is available", func() {
+
+				Specify("related images are loaded", func() {
+
+					sourceName := genName("catalog-")
+					packageName := "etcd-test"
+					image := "quay.io/olmtest/catsrc-update-test:related"
+
+					catalogSource := &v1alpha1.CatalogSource{
+						TypeMeta: metav1.TypeMeta{
+							Kind:       v1alpha1.CatalogSourceKind,
+							APIVersion: v1alpha1.CatalogSourceCRDAPIVersion,
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      sourceName,
+							Namespace: testNamespace,
+							Labels:    map[string]string{"olm.catalogSource": sourceName},
+						},
+						Spec: v1alpha1.CatalogSourceSpec{
+							SourceType: v1alpha1.SourceTypeGrpc,
+							Image:      image,
+						},
+					}
+
+					expectedRelatedImages := map[string]string{
+						"quay.io/coreos/etcd@sha256:3816b6daf9b66d6ced6f0f966314e2d4f894982c6b1493061502f8c2bf86ac84":          "",
+						"quay.io/coreos/etcd@sha256:49d3d4a81e0d030d3f689e7167f23e120abf955f7d08dbedf3ea246485acee9f":          "",
+						"quay.io/coreos/etcd-operator@sha256:c0301e4686c3ed4206e370b42de5a3bd2229b9fb4906cf85f3f30650424abec2": "",
+					}
+
+					catalogSource, err := crc.OperatorsV1alpha1().CatalogSources(catalogSource.GetNamespace()).Create(context.TODO(), catalogSource, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					defer func() {
+						err := crc.OperatorsV1alpha1().CatalogSources(catalogSource.GetNamespace()).Delete(context.TODO(), catalogSource.GetName(), metav1.DeleteOptions{})
+						Expect(err).NotTo(HaveOccurred())
+					}()
+
+					// Wait for package-server to be ready
+					err = wait.Poll(pollInterval, 1*time.Minute, func() (bool, error) {
+						ctx.Ctx().Logf("Polling package-server...")
+						_, err := pmc.OperatorsV1().PackageManifests(testNamespace).List(context.TODO(), metav1.ListOptions{})
+						if err == nil {
+							return true, nil
+						}
+						return false, nil
+					})
+					Expect(err).NotTo(HaveOccurred(), "package-server not available")
+
+					_, err = fetchCatalogSource(crc, catalogSource.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+					Expect(err).NotTo(HaveOccurred())
+
+					pm, err := fetchPackageManifest(pmc, testNamespace, packageName, packageManifestHasStatus)
+					Expect(err).NotTo(HaveOccurred(), "error getting package manifest")
+					Expect(pm).ShouldNot(BeNil())
+					Expect(pm.GetName()).Should(Equal(packageName))
+
+					relatedImages := pm.Status.Channels[0].CurrentCSVDesc.RelatedImages
+					Expect(len(relatedImages)).To(Equal(len(expectedRelatedImages)))
+
+					for _, v := range relatedImages {
```

Comment:
This could be:
```
Expect(relatedImages).To(ConsistOf([]string{
  "quay.io/coreos/etcd@sha256:3816b6daf9b66d6ced6f0f966314e2d4f894982c6b1493061502f8c2bf86ac84",
  "quay.io/coreos/etcd@sha256:49d3d4a81e0d030d3f689e7167f23e120abf955f7d08dbedf3ea246485acee9f",
  "quay.io/coreos/etcd-operator@sha256:c0301e4686c3ed4206e370b42de5a3bd2229b9fb4906cf85f3f30650424abec2"
}))
```

which would cover this for loop as well as the length assertion on L187.